### PR TITLE
Essentials.Gyroscope refactored to interface and singleton

### PR DESCRIPTION
**Pull request by moljac**
### Description of Change ###

Implements #

```csharp
	public interface IGyroscope
	{
		bool IsSupported { get; }

		void Start(SensorSpeed sensorSpeed);

		void Stop();
	}
```

### Additions made ###

* Adds 

### PR Checklist ###

- [ ] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [ ] Targets a single property for a single control (or intertwined few properties)
- [ ] Adds the property to the appropriate interface
- [ ] Avoids any changes not essential to the handler property
- [ ] Adds the mapping to the PropertyMapper in the handler
- [ ] Adds the mapping method to the WinUI, Android, iOS, and Standard aspects of the handler
- [ ] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [ ] Tags ported renderer methods with [PortHandler]
- [ ] Adds an example of the property to the sample project (MainPage)
- [ ] Adds the property to the stub class
- [ ] Implements basic property tests in DeviceTests

#### Does this PR touch anything that might affect accessibility?
- [ ] Does this PR introduce a new control? (If yes, add an example using SemanticProperties to the SemanticsPage)
- [ ] APIs that modify focusability?
- [ ] APIs that modify any text property on a control?
- [ ] Does this PR modify view nesting or view arrangement in anyway?
- [ ] Is there the smallest possibility that your PR will change accessibility? 
- [ ] I'm not sure, please help me

If any of the above checkboxes apply to your PR, then the PR will need to provide testing to demonstrate that accessibility still works.

### CODE REVIEW

File: src/Essentials/src/Gyroscope/Gyroscope.tizen.cs [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
-		internal static void PlatformStart(SensorSpeed sensorSpeed)
+		public void Start(SensorSpeed sensorSpeed)
 		{
 			DefaultSensor.Interval = sensorSpeed.ToPlatform();
 			DefaultSensor.DataUpdated += DataUpdated;
 			DefaultSensor.Start();
 		}
 
-		internal static void PlatformStop()
+		publicvoid Stop()
```
Comment (rmarinho) [inline src/Essentials/src/Gyroscope/Gyroscope.tizen.cs:21]:
```suggestion
		public void Stop()
```

**Review by rmarinho — DISMISSED** [review]
UWP missing 

```
         D:\a\_work\1\s\src\Essentials\src\Gyroscope\Gyroscope.uwp.cs(6,49): error CS0736: 'GyroscopeImplementation' does not implement instance interface member 'IGyroscope.Start(SensorSpeed)'. 'GyroscopeImplementation.Start(SensorSpeed)' cannot implement the interface member because it is static. [D:\a\_work\1\s\src\Essentials\src\Essentials.csproj]
```

**Comment by moljac** [comment]
Seems like packing and testing fails. What can I do?

File: src/Essentials/src/Gyroscope/Gyroscope.shared.cs [hunk 2]
```diff
@@ -13,10 +25,13 @@ public static partial class Gyroscope
 		/// <include file="../../docs/Microsoft.Maui.Essentials/Gyroscope.xml" path="//Member[@MemberName='IsMonitoring']/Docs" />
 		public static bool IsMonitoring { get; private set; }
```
Comment (rmarinho) [inline src/Essentials/src/Gyroscope/Gyroscope.shared.cs:26]: Shouldn't this be on the interface ?

Comment (Redth) [reply src/Essentials/src/Gyroscope/Gyroscope.shared.cs]: Fixed.

**Review by Redth — APPROVED** [review]

**Pull request merged**